src/coreclr/vm/jitinterface.cpp

```diff
@@ -9323,6 +9323,475 @@ CorInfoTypeWithMod CEEInfo::getArgType (
+                    }
+                }
+            }
+            else if (numIntroducedFields == 2)
+            {
+                FieldDesc *pFieldStart = pMethodTable->GetApproxFieldDescListRaw();
+
+                CorElementType fieldType = pFieldStart[0].GetFieldType();
+                if (CorTypeInfo::IsPrimitiveType_NoThrow(fieldType))
+                {
+                    if (fieldType == ELEMENT_TYPE_R4)
+                        size = STRUCT_FLOAT_FIELD_FIRST;
+                    else if (fieldType == ELEMENT_TYPE_R8)
+                        size = STRUCT_FIRST_FIELD_DOUBLE;
+                    else if (pFieldStart[0].GetSize() == 8)
+                        size = STRUCT_FIRST_FIELD_SIZE_IS8;
+
+                }
+                else if (fieldType == ELEMENT_TYPE_VALUETYPE)
+                {
+                    pMethodTable  = pFieldStart->LookupApproxFieldTypeHandle().GetMethodTable();
+                    if (pMethodTable->GetNumIntroducedInstanceFields() == 1)
+                    {
+                        size = getLoongArch64PassStructInRegisterFlags((CORINFO_CLASS_HANDLE)pMethodTable);
+                        if (size == STRUCT_FLOAT_FIELD_ONLY_ONE)
+                        {
+                            size = pFieldStart[0].GetSize() == 8 ? STRUCT_FIRST_FIELD_DOUBLE : STRUCT_FLOAT_FIELD_FIRST;
+                        }
+                        else if (size == STRUCT_NO_FLOAT_FIELD)
+                        {
+                            size = pFieldStart[0].GetSize() == 8 ? STRUCT_FIRST_FIELD_SIZE_IS8: 0;
+                        }
+                        else
+                        {
+                            size = STRUCT_NO_FLOAT_FIELD;
+                            goto _End_arg;
+                        }
+                    }
+                    else
+                    {
+                        size = STRUCT_NO_FLOAT_FIELD;
+                        goto _End_arg;
+                    }
+                }
+                else if (pFieldStart[0].GetSize() == 8)
+                    size = STRUCT_FIRST_FIELD_SIZE_IS8;
+
+                fieldType = pFieldStart[1].GetFieldType();
+                if (CorTypeInfo::IsPrimitiveType_NoThrow(fieldType))
+                {
+                    if (fieldType == ELEMENT_TYPE_R4)
+                        size = size & STRUCT_FLOAT_FIELD_FIRST ? (size ^ STRUCT_MERGE_FIRST_SECOND) : (size | STRUCT_FLOAT_FIELD_SECOND);
+                    else if (fieldType == ELEMENT_TYPE_R8)
+                        size = size & STRUCT_FLOAT_FIELD_FIRST ? (size ^ STRUCT_MERGE_FIRST_SECOND_8) : (size | STRUCT_SECOND_FIELD_DOUBLE);
+                    else if (pFieldStart[1].GetSize() == 8)
+                        size |= STRUCT_SECOND_FIELD_SIZE_IS8;
+                }
+                else if (fieldType == ELEMENT_TYPE_VALUETYPE)
+                {
+                    pMethodTable  = pFieldStart->LookupApproxFieldTypeHandle().GetMethodTable();
```

Comment:
LookupApproxFieldTypeHandle can return null. Is there anything that guarantees it won't return null in this case?

Should this rather use `GetFieldTypeHandleThrowing` that will load the types as necessary and never return null?

Reply:
Thanks.
You are right.
In fact, we had updated it by using a `if (pMethodTable)` and it works ok.

Reply:
> In fact, we had updated it by using a if (pMethodTable) and it works ok.

That's not a correct fix. `LookupApproxFieldTypeHandle` will return null when the type was not loaded before. Using `LookupApproxFieldTypeHandle` will lead to non-deterministic behavior - the behavior will depend on whether something else in the program happened to load the type or not.

You should change this code to call `GetFieldTypeHandleThrowing`.

Reply:
> > In fact, we had updated it by using a if (pMethodTable) and it works ok.
> 
> That's not a correct fix. `LookupApproxFieldTypeHandle` will return null when the type was not loaded before. Using `LookupApproxFieldTypeHandle` will lead to non-deterministic behavior - the behavior will depend on whether something else in the program happened to load the type or not.
> 
> You should change this code to call `GetFieldTypeHandleThrowing`.

Thanks a lot !!!
I will update it.